### DISCUSSION TITLE
Hexagon compilation problem

### DISCUSSION THREAD

**Discussion by XVilka** [rizinorg/rizin#1055] [Archive] [answered]
It's already known that Hexagon plugin often fails to compile with ASAN. Most recent compiler started to print this warning:
```
 ../librz/asm/arch/hexagon/hexagon_disas.c: In function ‘hexagon_disasm_instruction’:
../librz/asm/arch/hexagon/hexagon_disas.c:22:20: note: variable tracking size limit exceeded with ‘-fvar-tracking-assignments’, retrying without
   22 | NO_OPT_IF_ASAN int hexagon_disasm_instruction(ut32 hi_u32, HexInsn *hi, ut32 addr) {
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by ret2libc** [comment]
Does it still happen? I saw a lot of work on "hexagon", but I don't know what's going on there..

**Comment by wargio** [comment]
This might get fixed with https://github.com/rizinorg/rizin/pull/1338

**Comment by Rot127** [comment]
We should check this after https://github.com/rizinorg/rizin/pull/2548 has been merged.

**Comment by XVilka** [comment]
Haven't seen this happen anymore

**Comment by wargio** [comment, marked as answer]
it was fixed when @thestr4ng3r changed the way the disassembler works.
